syntax = "proto3";

package test.channel;

message Request {
  string name = 1;
}
  
message Response {
  string message = 1;
}

service Greeter {
    rpc HelloUnaryUnary (Request) returns (Response) {}
    rpc HelloUnaryStream (Request) returns (stream Response) {}
    rpc HelloStreamUnary (stream Request) returns (Response) {}
    rpc HelloStreamStream (stream Request) returns (stream Response) {}
}